### ISSUE TITLE
Coverity warnings in test_v2

### ISSUE THREAD

**Issue by rhc54** [openpmix/openpmix#2328]
```text
________________________________________________________________________________________________________
*** CID 373434:  API usage errors  (PRINTF_ARGS)
/test/test_v2/test_server.c: 874 in server_read_cb()
868                         strncpy(fence_sig[fence_idx][p].nspace, msg_hdr.procs[p].nspace, 255);
869                         // loop over all nodes to find which node has this proc
870                         for (n = 0; n < g_num_nodes; n++) {
871                             temp_nodeid = -1;
872                             // look for a match for this fence proc within the procs on this node
873                             for (i = 0; i < nodes[n].pmix_local_size; i++) {
   CID 373434:  API usage errors  (PRINTF_ARGS)
   Argument "nodes[n].pmix_local_size" to format specifier "%d" was expected to have type "int" but has type "unsigned long".
874                                 TEST_VERBOSE(("Node: %d Local size: %d rank: %d", n, nodes[n].pmix_local_size, nodes[n].pmix_rank[i]));
875                                 if (nodes[n].pmix_rank[i] == msg_hdr.procs[p].rank) {
876                                     temp_nodeid = nodes[n].pmix_nodeid;
877                                     TEST_VERBOSE(("Node: %d found for proc: %d", temp_nodeid, nodes[n].pmix_rank[i]));
878                                     break;
879                                 }

** CID 373433:  API usage errors  (PRINTF_ARGS)


________________________________________________________________________________________________________
*** CID 373433:  API usage errors  (PRINTF_ARGS)
/test/test_v2/server_callbacks.c: 124 in fencenb_fn()
118                              size_t ninfo, char *data, size_t ndata, pmix_modex_cbfunc_t cbfunc,
119                              void *cbdata)
120     {
121         size_t n;
122     
123         for (n = 0; n < nprocs; n++){
   CID 373433:  API usage errors  (PRINTF_ARGS)
   Argument "nprocs" to format specifier "%d" was expected to have type "int" but has type "unsigned long".
124             TEST_VERBOSE(("Participating in this fence on node: %d procs[%d].nspace: %s,"
125                           " procs[%d].rank: %d, nprocs: %d", my_server_id, n, procs[n].nspace, n,
126                           procs[n].rank, nprocs));
127         }
128     
129         /* commented out, for now, to allow for the possibility of timeouts */

** CID 373432:  API usage errors  (PRINTF_ARGS)


________________________________________________________________________________________________________
*** CID 373432:  API usage errors  (PRINTF_ARGS)
/test/test_v2/server_callbacks.c: 124 in fencenb_fn()
118                              size_t ninfo, char *data, size_t ndata, pmix_modex_cbfunc_t cbfunc,
119                              void *cbdata)
120     {
121         size_t n;
122     
123         for (n = 0; n < nprocs; n++){
   CID 373432:  API usage errors  (PRINTF_ARGS)
   Argument "n" to format specifier "%d" was expected to have type "int" but has type "unsigned long".
124             TEST_VERBOSE(("Participating in this fence on node: %d procs[%d].nspace: %s,"
125                           " procs[%d].rank: %d, nprocs: %d", my_server_id, n, procs[n].nspace, n,
126                           procs[n].rank, nprocs));
127         }
128     
129         /* commented out, for now, to allow for the possibility of timeouts */

** CID 373431:    (PRINTF_ARGS)


________________________________________________________________________________________________________
*** CID 373431:    (PRINTF_ARGS)
/test/test_v2/test_server.c: 772 in server_read_cb()
766         static pmix_proc_t fence_sig[PMIXT_MAX_FENCES][PMIXT_MAX_PROCS];
767         static fence_nodes_t fence_nodes[PMIXT_MAX_FENCES][PMIXT_MAX_NODES];
768         static size_t barrier_cnt = 0;
769         static size_t contrib_cnt = 0;
770         static size_t fence_buf_offset = 0;
771     
   CID 373431:    (PRINTF_ARGS)
   Argument "260UL" to format specifier "%d" was expected to have type "int" but has type "unsigned long".
772         TEST_VERBOSE(("sizeof(msg_hdr_t): %d, sizeof(pmix_proc_t): %d",
773                      sizeof(msg_hdr_t), sizeof(pmix_proc_t)));
774     
775         rc = read(server->rd_fd, &msg_hdr, sizeof(msg_hdr_t));
776         if (rc <= 0) {
777             return;
/test/test_v2/test_server.c: 772 in server_read_cb()
766         static pmix_proc_t fence_sig[PMIXT_MAX_FENCES][PMIXT_MAX_PROCS];
767         static fence_nodes_t fence_nodes[PMIXT_MAX_FENCES][PMIXT_MAX_NODES];
768         static size_t barrier_cnt = 0;
769         static size_t contrib_cnt = 0;
770         static size_t fence_buf_offset = 0;
771     
   CID 373431:    (PRINTF_ARGS)
   Argument "33320UL" to format specifier "%d" was expected to have type "int" but has type "unsigned long".
772         TEST_VERBOSE(("sizeof(msg_hdr_t): %d, sizeof(pmix_proc_t): %d",
773                      sizeof(msg_hdr_t), sizeof(pmix_proc_t)));
774     
775         rc = read(server->rd_fd, &msg_hdr, sizeof(msg_hdr_t));
776         if (rc <= 0) {
777             return;

** CID 373430:  Memory - illegal accesses  (STRING_NULL)
/test/test_v2/test_server.c: 928 in server_read_cb()


________________________________________________________________________________________________________
*** CID 373430:  Memory - illegal accesses  (STRING_NULL)
/test/test_v2/test_server.c: 928 in server_read_cb()
922                         resp_hdr.cmd = CMD_FENCE_COMPLETE;
923                         resp_hdr.size = fence_buf_offset;
924                         // is the fence_index field helping anything here?
925                         resp_hdr.fence_index = fence_idx;
926                         resp_hdr.nprocs = msg_hdr.nprocs;
927                         for (i = 0; i < msg_hdr.nprocs; i++) {
   CID 373430:  Memory - illegal accesses  (STRING_NULL)
   Passing unterminated string "msg_hdr.procs[i].nspace" to "strcpy", which expects a null-terminated string.
928                             strcpy(resp_hdr.procs[i].nspace, msg_hdr.procs[i].nspace);
929                             resp_hdr.procs[i].rank = msg_hdr.procs[i].rank;
930                         }
931                         server_send_msg(&resp_hdr, fence_buf, fence_buf_offset);
932                     }
933                     TEST_VERBOSE(

** CID 373429:  API usage errors  (PRINTF_ARGS)


________________________________________________________________________________________________________
*** CID 373429:  API usage errors  (PRINTF_ARGS)
/test/test_v2/test_fence_partial.c: 184 in main()
178                 }
179                 usecs_elapsed =
180                     ((end.tv_sec * 1000000 + end.tv_usec) - (start.tv_sec * 1000000 + start.tv_usec));
181                 secs_elapsed = (double) usecs_elapsed / 1E6;
182                 // secs_elapsed must be less than padded_fence_time since no one waits for rank 0
183                 if (secs_elapsed > padded_fence_time) {
   CID 373429:  API usage errors  (PRINTF_ARGS)
   Argument "this_proc.rank" to format specifier "%s" was expected to have type "char *" but has type "unsigned int".
184                     TEST_ERROR(("%s: PMIx_Fence timeout: Rank %d elapsed fence time: %lf exceeded cutoff of: %lf",
185                         this_proc.rank, this_proc.nspace, secs_elapsed, padded_fence_time));
186                     pmixt_exit(PMIX_ERR_TIMEOUT);
187                 }
188                 TEST_VERBOSE(("Completed node-dependent fence, participant rank: %d", this_proc.rank));
189                 // clean up procs array

** CID 373428:  API usage errors  (PRINTF_ARGS)


________________________________________________________________________________________________________
*** CID 373428:  API usage errors  (PRINTF_ARGS)
/test/test_v2/test_fence_partial.c: 184 in main()
178                 }
179                 usecs_elapsed =
180                     ((end.tv_sec * 1000000 + end.tv_usec) - (start.tv_sec * 1000000 + start.tv_usec));
181                 secs_elapsed = (double) usecs_elapsed / 1E6;
182                 // secs_elapsed must be less than padded_fence_time since no one waits for rank 0
183                 if (secs_elapsed > padded_fence_time) {
   CID 373428:  API usage errors  (PRINTF_ARGS)
   Argument "this_proc.nspace" to format specifier "%d" was expected to have type "int" but has type "char *".
184                     TEST_ERROR(("%s: PMIx_Fence timeout: Rank %d elapsed fence time: %lf exceeded cutoff of: %lf",
185                         this_proc.rank, this_proc.nspace, secs_elapsed, padded_fence_time));
186                     pmixt_exit(PMIX_ERR_TIMEOUT);
187                 }
188                 TEST_VERBOSE(("Completed node-dependent fence, participant rank: %d", this_proc.rank));
189                 // clean up procs array

** CID 373427:  API usage errors  (PRINTF_ARGS)


________________________________________________________________________________________________________
*** CID 373427:  API usage errors  (PRINTF_ARGS)
/test/test_v2/test_server.c: 824 in server_read_cb()
818             TEST_VERBOSE(("CMD_FENCE_CONTRIB req from server: %d cnt: %lu size: %d", msg_hdr.src_id,
819                           (unsigned long) contrib_cnt, msg_hdr.size));
820             // compare incoming fence to our existing fences
821             // assumption is that procs array will be constructed in the same order for every
822             // participating server
823             if (msg_hdr.nprocs > 0 && msg_hdr.procs[0].rank != PMIX_RANK_WILDCARD) {
   CID 373427:  API usage errors  (PRINTF_ARGS)
   Argument "msg_hdr.nprocs" to format specifier "%d" was expected to have type "int" but has type "unsigned long".
824                 TEST_VERBOSE(("Inside partial fence processing, source: %d, nprocs: %d"
825                              " fences_in_flight : %d", msg_hdr.src_id, msg_hdr.nprocs, fences_in_flight));
826                 // init fence_nodes array (must only happen first pass)
827                 if (fences_in_flight == 0) {
828                     for (i = 0; i < PMIXT_MAX_FENCES; i++) {
829                         for (n = 0; n < PMIXT_MAX_NODES; n++) {


_________________________________________
```

**Comment by rhc54** [comment]
@cpshereda Could you please resolve these?

**Comment by cpshereda** [comment]
Yes, I'll look at it today.

**Comment by rhc54** [comment]
closed by #2338
